### ISSUE TITLE
[Windows][Desktop][IME]: Focus is not reset after entering input in other textfields. Text gets duplicated. 

### ISSUE THREAD

**Issue by xyun52** [flutter/flutter#90503]
Switch to other TextInput when input is not complete. Input value confusion
![微信截图_20210922144753](https://user-images.githubusercontent.com/5417624/134297875-b3d4514f-4ac9-4528-8c04-28e93cd96007.png)

**Comment by darshankawar** [comment]
@xyun52 
Can you provide `flutter doctor -v`, clear steps to replicate along with actual vs expected behavior ?
Thanks.

**Comment by xyun52** [comment]
Flutter (Channel stable, 2.5.0, on Microsoft Windows [Version 10.0.17763.2183], locale zh-CN)
    • Flutter version 2.5.0 at C:\sdks\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4cc385b4b8 (2 weeks ago), 2021-09-07 23:01:49 -0700
    • Engine revision f0826da7ef
    • Dart version 2.14.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[X] Android toolchain - develop for Android devices
    • Android SDK at C:\Android\android-sdk
    X cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Enterprise 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise
    • Visual Studio Enterprise 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\X\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.26.0

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.17763.2183]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 93.0.4577.82

1. select first TextInput by mouse, input any character，don't input "space" or "enter".
2. select next TextInput by mouse, input any character，the input value confusion

**Comment by xyun52** [comment]
this is steps video and the video end new bug, long text select can't scrol

https://user-images.githubusercontent.com/5417624/134439819-fc0a99cd-a410-4eec-8df8-6e6d944e879c.mp4

l

**Comment by xyun52** [comment]
this is steps video and the video end new bug, long text select can't scrol

https://user-images.githubusercontent.com/5417624/134440041-8348763d-7ebc-46c1-b82e-1cdc567b3abc.mp4

l

**Comment by xyun52** [comment]
this is steps video and the video end new bug, long text select can't scroll

https://user-images.githubusercontent.com/5417624/134440827-24788c6c-068d-44e4-8963-b837a7602802.mp4

**Comment by darshankawar** [comment]
@xyun52 
Tried following your steps to replicate on Windows 10 machine and running a sample app with 2 text fields on desktop. Used mouse to input char / text but didn't observe the behavior you are reporting. Also tried inputing localized char and using soft keyboard, but no issues faced.

![char](https://user-images.githubusercontent.com/67046386/134474221-b5f48209-6ed8-421f-a7d7-5b95bb4bc9cf.png)
![char_in](https://user-images.githubusercontent.com/67046386/134474223-5353808a-24b9-402a-9b13-84a97b865bb9.png)


<details>
<summary> flutter doctor -v </summary>

```



[√] Flutter (Channel stable, 2.5.1, on Microsoft Windows [Version 10.0.19042.1237], locale en-US)
    • Flutter version 2.5.1 at D:\FlutterSDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (4 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Win\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\Win\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (4 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm    • Android 8.1.0 (API 27)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [Version 10.0.19042.1237]
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 93.0.4577.82
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 93.0.961.52

• No issues found!

```

</details>

**Comment by xyun52** [comment]
@darshankawar 
You may need to use other input methods, such as Chinese input method.

![image](https://user-images.githubusercontent.com/5417624/134481213-6c5734a9-17cc-4398-9c57-fdc6a451a93a.png)

**Comment by darshankawar** [comment]
@xyun52 
Can you try the solution mentioned in [this](https://github.com/flutter/flutter/issues/77586#issuecomment-800995554) issue comment and see if it helps ?

**Comment by xyun52** [comment]
@darshankawar 
Can't solve the problem i encountered

**Comment by xyun52** [comment]
The beta channel has the same problem

[√] Flutter (Channel beta, 2.6.0-5.2.pre, on Microsoft Windows [Version 10.0.17763.2183], locale zh-CN)
    • Flutter version 2.6.0-5.2.pre at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 400608f101 (8 days ago), 2021-09-15 15:50:26 -0700
    • Engine revision 1d521d89d8
    • Dart version 2.15.0 (build 2.15.0-82.2.beta)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

**Comment by darshankawar** [comment]
Ok I was able to replicate and see the same behavior as OP using latest master and stable. 

Steps to replicate:

1. Use Chinese Simplied input keyboard.
2. In Textfield, input ab
3. In next textfield, input same text (ab) which shows duplicated text from textfield 1.

![test](https://user-images.githubusercontent.com/67046386/134630532-a77bad65-e9c6-4d4f-b1c3-811c81b1e787.png)


<details>
<summary> code sample </summary>

```

body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              TextField(
                decoration: InputDecoration(
                    hintText: 'enter'
                ),
              ),
              TextField(
                decoration: InputDecoration(
                    hintText: 'enter'
                ),
              )
            ],
          )
        )

```

</details>


<details>
<summary> flutter doctor -v </summary>

```

[√] Flutter (Channel master, 2.6.0-6.0.pre.198, on Microsoft Windows [Version 10.0.19042.1237], locale en-US)
    • Flutter version 2.6.0-6.0.pre.198 at D:\FlutterSDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d32960d03a (58 minutes ago), 2021-09-24 01:43:07 -0400
    • Engine revision a5b860a265
    • Dart version 2.15.0 (build 2.15.0-142.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Win\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\Win\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19042.1237]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 93.0.4577.82
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 93.0.961.52

• No issues found!

[√] Flutter (Channel stable, 2.5.1, on Microsoft Windows [Version 10.0.19042.1237], locale en-US)
    • Flutter version 2.5.1 at D:\FlutterSDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (4 days ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Win\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.60.1)
    • VS Code at C:\Users\Win\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (4 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm    • Android 8.1.0 (API 27)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [Version 10.0.19042.1237]
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 93.0.4577.82
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 93.0.961.52

• No issues found!

```

</details>

**Comment by xyun52** [comment]
@darshankawar 
Our customers often click and switch when editing large forms. This problem is very serious and I hope it can be fixed as soon as possible.

**Comment by cbracken** [comment]
This was fixed in flutter/engine#29146.

**Comment by dkwingsmt** [comment]
I filed xyun52's issue in a separate one https://github.com/flutter/flutter/issues/92050
